Rubberduck.Parsing/Grammar/PartialExtensions/ConstSubStmtContext.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Antlr4.Runtime.Misc;
+using Rubberduck.Parsing.Grammar.Abstract;
+using Rubberduck.Parsing.Symbols;
+
+namespace Rubberduck.Parsing.Grammar
+{
+    public partial class VBAParser
+    {
+        public partial class ConstSubStmtContext : IIdentifierContext, IAnnotatedContext
```

Comment:
I noticed the same thing, and it tickles. I think the problem is that the nodes need to implement the interfaces, since they are being cast to these interfaces in a couple of places.. and implementing CPA traversal will be much easier if nodes can be cast to (or filtered by) the CPA interfaces (e.g. `IExecutableNode`) they're implementing.

For example there could easily be an `ExecutableNodeImpl` class:

```csharp
        public class ExecutableNodeImpl : IExecutableNode
        {
            private readonly IDictionary<IExecutionContext, bool> _hasExecuted
                = new Dictionary<IExecutionContext, bool>();

            public bool HasExecuted(IExecutionContext context) 
                => _hasExecuted.TryGetValue(context, out var value) && value;

            public void Execute(IExecutionContext context)
                => _hasExecuted[context] = true;
        }
```
And then the `GoToStmtContext` extension could do this:

```csharp
    private readonly ExecutableNodeImpl _executableNode = new ExecutableNodeImpl();
```

But then, it's no longer implementing `IJumpNode : IExecutableNode`, so it would need to define the members anyway.

I'm concluding that as long as we want/need tree nodes to implement interfaces, we'll have to put up with stupid-simple, but repetitive code in all the partial extension classes.